framework/src/main/java/org/checkerframework/framework/util/typeinference/constraint/AFReducingVisitor.java

```diff
@@ -309,7 +309,7 @@ public Void visitIntersection_Declared(
         // at least one of the intersection bound types must be convertible to the param type
         final AnnotatedDeclaredType subtypeAsParam =
                 AnnotatedTypes.castedAsSuper(typeFactory, subtype, supertype);
-        if (subtypeAsParam != null && !subtypeAsParam.equals(subtype)) {
+        if (subtypeAsParam != null && !subtypeAsParam.equals(supertype)) {
```

Comment:
This one wasn't actually found by the Interning Checker.  (I found it via an IntelliJ analysis.)